src/OpenSearch/Handlers/SigV4Handler.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSearch\Handlers;
+
+use Aws\Credentials\CredentialProvider;
+use Aws\Signature\SignatureV4;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Uri;
+use OpenSearch\ClientBuilder;
+use Psr\Http\Message\RequestInterface;
+use RuntimeException;
+
+class SigV4Handler
```

Comment:
What about adding a new method to the ClientBuilder like `enableSig4Authentification(?string $accessKey= null, ?string $accessSecret = null)`

Reply:
I added a couple methods to the ClientBuilder that mimic how the AWS SDK accepts credentials.